lib/stud/task.rb

```diff
@@ -1,8 +1,21 @@
 require "thread"
+require "stud/interval"
 
 module Stud
+
+  # A Task spawns a thread to execute the given block. execution completion and result retrieval is
+  # done using the Task#wait method. A Task in run once and the thread exists upon block completion.
+  # A task and its underlying thread are not reusable.
+  #
+  # Task does not provide a mean to force-interrupt a running task, it only provides the #stop!
+  # method to signal the task for a top request. The task or code block can use the #stop? method
+  # to check for a stop request. Note that the #stop! and #stop? methods are thread safe.
   class Task
+    attr_reader :thread
+
     def initialize(*args, &block)
+      # provide access to the underlying thread if ever needed.
```

Comment:
dear god. for some reason I originally mis placed the attr_reader in the inititalize methods and obviously moved it because it's not valid but the comment did  not follow :P